fe/fe-core/src/main/java/org/apache/doris/analysis/SelectStmt.java

```diff
@@ -444,13 +444,18 @@ public void analyze(Analyzer analyzer) throws UserException {
             }
         }
         if (groupByClause != null && groupByClause.isGroupByExtension()) {
+            ArrayList<Expr> aggFnExprList = new ArrayList<>();
             for (SelectListItem item : selectList.getItems()) {
-                if (item.getExpr() instanceof FunctionCallExpr && item.getExpr().fn instanceof AggregateFunction) {
+                aggFnExprList.clear();
+                getAggregateFnExpr(item.getExpr(), aggFnExprList);
+                if (!aggFnExprList.isEmpty()) {
```

Comment:
this is more clear and simple:
`for (Expr aggFnExpr : aggFnExprList) {
    for (Expr expr : groupByClause.getGroupingExprs()) {
        ....
    }
}`

Reply:
done